termwiz/src/render/terminfo.rs

```diff
@@ -50,13 +50,17 @@ impl TerminfoRenderer {
     #[cfg_attr(feature = "cargo-clippy", allow(clippy::cognitive_complexity))]
     fn flush_pending_attr<W: RenderTty + Write>(&mut self, out: &mut W) -> Result<()> {
         macro_rules! attr_on {
-            ($cap:ident, $sgr:expr) => {
-                if let Some(attr) = self.get_capability::<cap::$cap>() {
+            ($cap:ident, $sgr:expr) => {{
+                let mut cap = self.get_capability::<cap::$cap>();
+                if self.caps.force_terminfo_sgr_fallback() {
+                    cap = None;
+                }
```

Comment:
Minor stylistic nit: I would phrase it this way around to avoid the less-idiomatic use of `mut cap`; same suggestion for the code around `cap::SetAttributes` below:

```suggestion
                let cap = if self.caps.force_terminfo_sgr_fallback() {
                   None
                } else {
                    self.get_capability::<cap::$cap>()
                }
```